src/Limenius/ReactRenderer/Context/GenericContextProvider.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Limenius\ReactRenderer\Context;
+
+/**
+ * Class ContextProvider
+ *
+ * Extracts context information from a URI path.
+ * https://en.wikipedia.org/wiki/Uniform_Resource_Identifier
+ */
+class GenericContextProvider implements ContextProviderInterface
+{
+    private $regex = '/(?<scheme>https?):\/\/(?<host>[^\/|:]*)(?::(?<port>\d*))?(?<uri>(?<base>\/?[^\.]*\.[^\/\?]*)?(?<path>[^?\.*]*)\??(?<search>.*))/';
```

Comment:
Why a regex instead of PHP's [parse_url()](https://www.php.net/manual/en/function.parse-url.php)?

Reply:
Thats three hours of my life I'll never get back - wasn't aware of that method, I'll update the PR later this evening. Thanks